web_src/js/features/repo-settings-branches.ts

```diff
@@ -0,0 +1,33 @@
+import {createSortable} from '../modules/sortable.ts';
+import {POST} from '../modules/fetch.ts';
+import {showErrorToast} from '../modules/toast.ts';
+
+export function initRepoBranchesSettings() {
+  const protectedBranchesList = document.querySelector('#protected-branches-list');
+  if (!protectedBranchesList) return;
+
+  createSortable(protectedBranchesList, {
+    handle: '.drag-handle',
+    animation: 150,
+
+    onEnd: () => {
+      (async () => {
+        const newOrder = Array.from(protectedBranchesList.children, (item) => {
+          const id = item.getAttribute('data-id');
+          return parseInt(id);
+        }).filter((id) => !Number.isNaN(id));
```

Comment:
```suggestion
        const newOrder = Array.from(queryElemChildren(protectedBranchesList, '.item[data-id]'), (el) => el.getAttribute('data-id'));
```

But why it is called `newOrder`? The code was copied&pasted from some unknown place?

Reply:
to be eplizite the initial frontend TS code was mate by claude.ai feeded by our frontend code and tasked to adopt what we do in web_src/js/features/repo-issue-list.ts but for the new endpoint & template.

so you could say unknown ...
... but It is already heavily modified and fixed by me so not sure what remains ... beside the naming

(I'm bad at naming so if there is a plausible name i just let it stay)

what would you suggest?

Reply:
I think the simplified code is better, there is no "defensive programming" here. We could clearly query all items with "data-id" attribute.

Meanwhile, `newOrder` could be something like `itemIds` since it is for item IDs.

Reply:
```
const itemElems = queryElemChildren(protectedBranchesList, '.item[data-id]');
const itemIds = Array.from(itemElems, (el) => el.getAttribute('data-id'));
```

Reply:
your suggestion did not work ... so i simplified it on your first comment

Reply:
Why?

Reply:
Why not work?

```
<div id="test-list">
	<div class="item" data-id="1"></div>
	<div class="item"></div>
	<div class="item" data-id="2"></div>
</div>


const protectedBranchesList = document.querySelector('#test-list');
const itemElems = queryElemChildren(protectedBranchesList, '.item[data-id]');
const itemIds = Array.from(itemElems, (el) => el.getAttribute('data-id'));
console.log(itemIds);
```

![image](https://github.com/user-attachments/assets/f3f73446-08c9-42d0-9226-f8b5970e5cbd)

![image](https://github.com/user-attachments/assets/7654b3d8-d481-4588-8d46-1e2fda79f0e2)

Reply:
forgot the import ...